### ISSUE TITLE
Macro benchmarks

### ISSUE THREAD

**Issue by FrostMegaByte** [pythonJaRvis/pythonJaRvis.github.io#6]
In your paper and code, it is mentioned that the macro benchmarks consist of the following GitHub projects:
- bpytop
- furl
- rich-cli
- sqlparse
- sshtunnel
- TextRank4ZH

However, these projects will get updated and then the benchmark results don't match anymore...
Are your macro benchmarks related to specific commit hashes from those repositories? Or is there another way to keep the repositories consistent, so I can run the same macro benchmarks as you did?

**Comment by pythonJaRvis** [comment]
yes, there is no rules for macro benchmark, you can test any project

**Comment by Mryanhehe** [comment]
"projects will get updated and the benchmark results don't match anymore". you are right, this groundtruth is for my paper, so we don't think about what they will be in the future. You could test these projects on python3.10, it may be different.

**Comment by FrostMegaByte** [comment]
@Mryanhehe Thanks for your answer. I understand that the groundtruth is for your paper, but this is based on projects which will change. For the reproducibility of your paper, I'd like to know the **commit hash of each of the 6 repositories** that you ran the callgraph generator on.

**Project's that have not been updated as of late:**
- bpytop _(Commit hash: [ac0f666](https://github.com/aristocratos/bpytop/commit/ac0f6660be0ad87b23ecb7f371a4a737e1bf017f))_
- furl _(Commit hash: [7748462](https://github.com/gruns/furl/commit/774846234ff803606fdd289a7549f9b50b2b3677))_
- rich-cli _(Commit hash: [0721cb1](https://github.com/Textualize/rich-cli/commit/0721cb196661687d8ecd9af4baf48576b240c978))_
- TextRank4ZH_(Commit hash: [fb13396](https://github.com/letiantian/TextRank4ZH/commit/fb1339620818a0b0c16f5613ebf54153faa41636))_

**Projects that have been updated:**
- sqlparse
-  sshtunnel
